### PR TITLE
Enable offloading for `numba.njit` in `dpctl.device_context`

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
Relates to https://github.com/IntelPython/dpctl/pull/678.

Tasks:
- [x] docs
- [x] changelog
- [x] `dppy.offload_to_sycl_device` reuses `dpctl.device_context`
- [x] replace `dppy.offload_to_sycl_device` with `dpctl.device_context` in all tests
- [x] dpctl >=0.12 or >=0.11.2

### CODE REVIEW

File: numba_dppy/tests/test_with_context.py [hunk 1]
```diff
@@ -16,14 +16,52 @@
...
+def scenario(filter_str, context):
+    @njit
+    def func(a, b):
+        return a + b
+
+    a = np.ones((64), dtype=np.int32)
+    b = np.ones((64), dtype=np.int32)
+
+    device = dpctl.SyclDevice(filter_str)
+    with context(device):
```
Comment (reazulhoque) [inline numba_dppy/tests/test_with_context.py:53]: How are we testing if `@njit` has actually been offloaded and did not fallback?

Comment (PokhodenkoSA) [reply numba_dppy/tests/test_with_context.py:53]: It is tested in `assert_auto_offloading` call. See below.

File: numba_dppy/tests/test_with_context.py [hunk 2]
```diff
@@ -16,14 +16,52 @@
...
+        func(a, b)
+
+
+@pytest.mark.parametrize("filter_str", filter_strings)
+@pytest.mark.parametrize("context", [
+    dppy.offload_to_sycl_device,
+    dpctl.device_context,
+])
+def test_dpctl_device_context_affects_numba_pipeline(filter_str, context):
+    with assert_auto_offloading():
```
Comment (PokhodenkoSA) [inline numba_dppy/tests/test_with_context.py:63]: This context enable debugging and wait for offloading diagnostic messages.

**Comment by PokhodenkoSA** [comment]
Documentation update is not necessary. Because we have example with `dpctl.device_context` on the main page.

**Comment by PokhodenkoSA** [comment]
Ready to be reviewed.
Waiting to be merged after https://github.com/IntelPython/dpctl/pull/678.

**Comment by PokhodenkoSA** [comment]
Merging based on public CI and local testing.

**Pull request merged**